src/Raven.Client/Http/HttpCache.cs

```diff
@@ -278,5 +282,15 @@ public void Dispose()
             }
             _unmanagedBuffersPool.Dispose();
         }
+
+        public void LowMemory()
+        {
+            _isFreeSpaceRunning.Raise();
+            FreeSpace();
```

Comment:
For example, we have concurrency between calling `Raise` here and the call from `Set`.
Both will call `LowerOrDie` and one of them will die

Reply:
Done. Both will check for .Raise before going into the function. Whoever raises it first will also go into the function and end up lowering it.